packages/components/src/toggle-group-control/stories/index.tsx

```diff
@@ -135,3 +139,142 @@ Deselectable.args = {
 	...WithIcons.args,
 	isDeselectable: true,
 };
+
+// TODO: remove before merging
+export const DoubleToggles: ComponentStory<
+	typeof ToggleGroupControl
+> = () => {
+	const aligns = [ 'Left', 'Center', 'Right' ];
+	const quantities = [ 'One', 'Two', 'Three', 'Four' ];
+
+	const [ alignState, setAlignState ] = useState< string | undefined >();
+	const [ quantityState, setQuantityState ] = useState<
+		string | undefined
+	>();
+
+	return (
+		<div>
+			<ToggleGroupControl
+				onChange={ ( value ) => setAlignState( value as string ) }
+				value={ alignState }
+				label={ 'Pick an alignment option' }
+			>
+				{ aligns.map( ( key ) => (
+					<ToggleGroupControlOption
+						key={ key }
+						value={ key }
+						label={ key }
+					/>
+				) ) }
+			</ToggleGroupControl>
+			<Button
+				onClick={ () => setAlignState( undefined ) }
+				variant="tertiary"
+			>
+				Reset
+			</Button>
+
+			<ToggleGroupControl
+				onChange={ ( value ) => setQuantityState( value as string ) }
+				value={ quantityState }
+				label={ 'Pick a quantity' }
+			>
+				{ quantities.map( ( key ) => (
+					<ToggleGroupControlOption
+						key={ key }
+						value={ key }
+						label={ key }
+					/>
+				) ) }
+			</ToggleGroupControl>
+			<Button
+				onClick={ () => setQuantityState( undefined ) }
+				variant="tertiary"
+			>
+				Reset
+			</Button>
+		</div>
+	);
+};
+
+// TODO: Remove before merging as well.
+const ExampleSlotFill = createSlotFill( 'Example' );
+
+const Slot = () => {
+	const motionContextValue = useContext( MotionContext );
+
+	// Forwarding the content of the slot so that it can be used by the fill
+	const fillProps = useMemo(
+		() => ( {
+			forwardedContext: [
+				[ MotionContext.Provider, { value: motionContextValue } ],
+			],
+		} ),
+		[ motionContextValue ]
+	);
+
+	return <ExampleSlotFill.Slot bubblesVirtually fillProps={ fillProps } />;
+};
+
+const Fill = ( { children }: { children: React.ReactNode } ) => {
+	const innerMarkup = <>{ children }</>;
+
+	return (
+		<ExampleSlotFill.Fill>
+			{ ( fillProps: {
+				forwardedContext?: [
+					React.Context< any >[ 'Provider' ],
```

Comment:
This type is too constraining, we also have `<StyleProvider document={ ... }>` which doesn't satisfy this.

`forwardedContext` is an array of entries like:
```ts
type Entry = [ Provider<Props>, Props ]
```
where `Provider` is a React component that accepts `Props` and also must accept `children`. Therefore:
```ts
type Provider<Props> = React.Component<PropsWithChildren<Props>>;
```
That's all what `forwardedContext` must satisfy, nothing more.

Reply:
`React.Component<React.PropsWithChildren<any>>` didn't work for me — the line below where I rendered `<Provider { ...props }>{ inner }</Provider>` was causing an error:

`JSX element type 'Provider' does not have any construct or call signatures.ts(2604)`


Changing the type to `React.FunctionComponent<React.PropsWithChildren<any>>` makes TSC happy.

Also worth noting that these extra storybook examples will be removed before merging the PR (which makes this specific type not super important), although I appreciate the review comment — learning every day!

Reply:
`React.ComponentType` is even better, allows all kinds of components. That's what I should originally have written, `React.Component` is a different thing 🙂 

One day the context forwarding will be typed, so the discussion is relevant beyond one storybook.

The type for the props is still wrong: it requires `value`, but for `StyleProvider` valid props are `{ document }`. Also it would be nice if the provider and props types were to match each other. Currently it's `any` everywhere, that's not real typing 🙂

Reply:
Something like this perhaps? https://github.com/WordPress/gutenberg/pull/50278/commits/edb79a5500a0ea8608a2eaf66369383be2ccfaf9

Reply:
Yes, this is better :+1: